src/app/util/attribute-table.h

```diff
@@ -17,14 +17,54 @@
 
 #pragma once
 
+#include <app/ConcreteAttributePath.h>
 #include <app/util/af-types.h>
 #include <app/util/attribute-metadata.h>
 #include <lib/core/DataModelTypes.h>
 #include <protocols/interaction_model/StatusCode.h>
 
+struct EmberAfWriteDataInput
+{
+    // where the input data originates from. NOTE: at this time there is no information regarding
+    // input buffer size and it is assumed "correct" for the given data type
+    uint8_t * dataPtr;
```

Comment:
It should have been, however existing code takes non-const so I am concerned about blast radius:
https://github.com/project-chip/connectedhomeip/blob/master/src/app/util/attribute-table.h#L54

I think if we want to switch to const (which would be more correct) we should do it as a separate PR.

It probably also will not be as easy because ember has a `ReadOrWrite` can that can do either side, so then the pointer passed to it must be non-const because it may be a read:

https://github.com/project-chip/connectedhomeip/blob/master/src/app/util/attribute-table.cpp#L439

... in the end, it seems to me this is ugly but fixing it is rough.

Reply:
I added a comment about this oddity. I think fix should not be part of this PR - existing methods were already taking in non-const, so I did not change that.